### PR TITLE
Adapt dev docs to include backporting information

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This pull request adapts the Developer How To to include the information on backporting. 
It shows the different ways that the developer can backport a pull request, whether it has been merged or not.

### CODE REVIEW

File: docs/development/dev_howto.rst [hunk 1]
```diff
@@ -532,6 +532,21 @@ Here's to commands to check for and fix this (see `here <http://stackoverflow.co
     $ git status
     $ cd astropy_helpers && git checkout -- . && cd ..
 
+Making a pull request that requires backport
+++++++++++++++++++++++++++++++++++++++++++++
+
+Some PRs will need to be backported to earlier versions of the code, for example
```
Comment (registerrier) [inline docs/development/dev_howto.rst:538]:
```suggestion
Some PRs will need to be backported to specific maintenance branches, for example
```

File: docs/development/dev_howto.rst [hunk 2]
```diff
@@ -532,6 +532,21 @@ Here's to commands to check for and fix this (see `here <http://stackoverflow.co
...
+bug fixes or correcting typos in doc-strings. There are a few ways this can be done,
+one of which involves the `meeseeksmachine <https://github.com/meeseeksmachine>`__.
+
+1. Add the backport label to automatically backport your PR. i.e. "``backport-v1.1.v``"
+
+2. If you forgot, on your merged PR make the comment: "``@meeseeksdev backport to [BRANCHNAME]``"
+
+3. If this does not work automatically, a set of instructions will be given to you in the next comment. This involves using the "``cherry-pick``" command. See `here <https://docs.astropy.org/en/latest/development/releasing.html#backporting-fixes-from-main>`__ for information.
```
Comment (bkhelifi) [inline docs/development/dev_howto.rst:546]:
```suggestion
3. If this does not work automatically, a set of instructions will be given to you in the next comment. This involves using the "``cherry-pick``" git command. See `here <https://docs.astropy.org/en/latest/development/releasing.html#backporting-fixes-from-main>`__ for information.
```

**Review by registerrier — DISMISSED** [review]
Thanks @Astro-Kirsty ! I have left a few inline comments.

**Comment by Astro-Kirsty** [comment]
Thanks @bkhelifi and @registerrier !

**Pull request merged**